lib/image-stream/parse-partitions.js

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const Bluebird = require('bluebird');
+const MBR = require('mbr');
+const GPT = require('gpt');
+
+const MAX_BYTES = 8192;
```

Comment:
Because the first time around I wasn't sure how many bytes it'd need from ISOs, the change to 8K is based on the max block size we're checking for, which is 4096. But now that I'm thinking about it, this would need changing to `MAX_BLOCK_SIZE * 34` because the GPT partition entries go from LBA 2 -> LBA34 (the first 4 at LBA2, 5-128 in LBA 3-34)